src/cli.rs

```diff
@@ -508,13 +514,16 @@ fn get_actual_app_id(
     path: &Option<String>,
 ) -> Result<String, CommandError> {
     match app_id {
+        Some(id) if id.is_empty() => {
+            Err(CommandError::EmptyAppId)
+        },
         Some(id) => Ok(id.clone()),
         None => {
             let manifest =
                 EdgeAppManifest::new(transform_edge_app_path_to_manifest(path).as_path()).unwrap();
             match manifest.app_id {
-                Some(id) => Ok(id),
-                None => {
+                Some(id) if !id.is_empty() => Ok(id.clone()),
+                _ =>  {
```

Comment:
Minor: None is ok here and more readable - what else could it be?)

Reply:
It can be Some(id) and empty :)